src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs

```diff
@@ -509,6 +515,60 @@ public Task<Block> GetAsync(uint256 hash)
             return task;
         }
 
+        /// <summary>
+        /// Get multiple blocks from the database by their block hashes.
+        /// </summary>
+        /// <param name="hashes">The block hashes.</param>
+        public Task<List<Block>> GetBlocksAsync(List<uint256> hashes)
+        {
+            Guard.NotNull(hashes, nameof(hashes));
+            this.logger.LogTrace("({0}:{1})", nameof(hashes.Count), hashes.Count);
+
+            Task<List<Block>> task = Task.Run(() =>
+            {
+                this.logger.LogTrace("()");
+
+                var results = new Dictionary<uint256, Block>();
+
+                using (DBreeze.Transactions.Transaction transaction = this.DBreeze.GetTransaction())
+                {
+                    transaction.ValuesLazyLoadingIsOn = false;
+
+                    // Access hash keys in sorted order.
+                    var byteListComparer = new ByteListComparer();
+                    var keys = hashes.Select(hash => (hash, hash.ToBytes())).ToList();
+
+                    keys.Sort((key1, key2) => byteListComparer.Compare(key1.Item2, key2.Item2));
+
+                    foreach (var key in keys)
+                    {
+                        Row<byte[], Block> blockRow = transaction.Select<byte[], Block>("Block", key.Item2);
+                        if (blockRow.Exists)
+                        {
+                            results[key.Item1] = blockRow.Value;
+                            this.PerformanceCounter.AddRepositoryHitCount(1);
+
+                            this.logger.LogTrace("(-):{0}={1}", key.Item1, blockRow.Value);
+                        }
+                        else
+                        {
+                            this.PerformanceCounter.AddRepositoryMissCount(1);
+
+                            this.logger.LogTrace("(-):{0}=[NO_BLOCK]", key.Item1);
```

Comment:
We are not exiting the method here, so just

```
this.logger.LogTrace("Block hash '{0}' not found in the store.", key.Item1);
```

and same for above, when we found it

```
this.logger.LogTrace("Block hash '{0}' loaded from the store.", key.Item1);
```

i don't think we have reasonable Block.ToString overload, so we won't log the block

Reply:
Done.